bitcoin/script/script.h

```diff
@@ -587,4 +573,27 @@ struct CScriptWitness
 
 bool CheckMinimalPush(const std::vector<unsigned char>& data, opcodetype opcode);
 
+/** Build a script by concatenating other scripts, or any argument accepted by CScript::operator<<. */
+template<typename... Ts>
+CScript BuildScript(Ts&&... inputs)
+{
+    CScript ret;
+
+    ([&] (Ts&& input) constexpr {
+        if constexpr (std::is_same_v<std::remove_reference_t<Ts>, CScript>) {
+            // If it is a CScript, extend ret with it. If ret is empty, move or copy it instead.
+            if (ret.empty()) {
```

Comment:
If you're concerned about the cost of the emptiness check here, it could be replaced with a `if (cnt++ == 0) {` instead, and an `int cnt{0};` outside the loop body. I suspect that in that case the compiler will actually realize the cnt can only be 0 on the first iteration, and not actually emit the check and conditional.